appleauth/auth_source.go

```diff
@@ -0,0 +1,153 @@
+package appleauth
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver/devportalservice"
+)
+
+// Source returns a specific kind (Apple ID/API Key) Apple authentication data from a specific source (Bitrise Service, manual input)
+type Source interface {
+	Fetch(connection *devportalservice.AppleDeveloperConnection, inputs Inputs) (*Credentials, error)
+	Description() string
+	RequiresConnection() bool
+}
+
+// ConnectionAPIKeySource provides API Key from Bitrise Service
+type ConnectionAPIKeySource struct{}
+
+// InputAPIKeySource provides API Key from manual input
+type InputAPIKeySource struct{}
+
+// ConnectionAppleIDSource provides Apple ID from Bitrise Service
+type ConnectionAppleIDSource struct{}
+
+// InputAppleIDSource provides Apple ID from manual input
+type InputAppleIDSource struct{}
+
+// Description ...
+func (*ConnectionAPIKeySource) Description() string {
+	return "Connected Apple Developer Portal Account for App Store Connect API found"
```

Comment:
Similar as above: `API key-based Bitrise Apple Developer Connection found`

Reply:
Some leftover `App Store Connect API Key` (should be `API Key`):
- https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver/pull/92/files#diff-03e3b537af3f24247110748c96acf8f29e29bb4d2f10793ec7913b6a64ef0251R15
- https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver/pull/92/files#diff-03e3b537af3f24247110748c96acf8f29e29bb4d2f10793ec7913b6a64ef0251R68
-